h/util/metrics.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import unicode_literals
+import newrelic.agent
+
+
+def record_search_query_param_usage(params, separate_replies, record_param=newrelic.agent.add_custom_parameter):
+    # Record usage of search params and associate them with a transaction.
+    keys = [
+        # Record usage of inefficient offset and it's alternative search_after.
+        "offset",
+        "search_after",
+        "sort",
+        # Record usage of url/uri (url is an alias of uri).
+        "url",
+        "uri",
+        # Record usage of tags/tag (tags is an alias of tag).
+        "tags",
+        "tag",
+        # Record group and user-these help in identifying slow queries.
+        "group",
+        "user",
+        # Record usage of wildcard feature.
+        "wildcard_uri"
+    ]
+
+    for k in keys:
+        if k in params:
+            # The New Relic Query Language does not permit _ at the begining
+            # and offset is a reserved key word.
+            record_param("es_{}".format(k), params[k])
+
+    # Record usage of _separate_replies which will help distinguish client calls
+    # for loading the sidebar annotations from other api calls.
+    record_param("es_{}".format("_separate_replies"), separate_replies)
+
+
+def add_custom_parameter(name, param):
```

Comment:
We discussed on a Slack call: possibly renaming this function to be a tad more generic for external callers (but leaving it public).